### PR TITLE
Travis and Codecov integration.

### PR DESCRIPTION
**Pull request by muzaffaryousaf**
General cleanup to test out Travis integration.

### CODE REVIEW

**Comment by muzaffaryousaf** [comment]
@yro Quick look ?

**Comment by muzaffaryousaf** [comment]
@yro Can you please give it a careful look, i've added codecov and fixed few quality & imports issue. Not sure if there's some hidden magic which i'm not able to see.

File: .coveragerc [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# .coveragerc for edx-video-pipeline
+[run]
+data_file = reports/.coverage
+source =
+    VEDA_OS01
```
Comment (muhammad-ammar) [inline .coveragerc]: don't we need coverage of `VEDA` source directory?

File: .travis.yml [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
   - "2.7"
+
 sudo: required
 
-install: pip install -r requirements.txt
-install: pip install pep8
+install:
+  - pip install -r requirements.txt
+  - pip install codecov
+  - pip install pep8
```
Comment (muhammad-ammar) [inline .travis.yml]: Should we move these last two installs in a file like `test-requirements.txt`?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,17 @@
+PACKAGES = VEDA VEDA_OS01 control frontend youtube_callback scripts
+
+validate: test ## Run tests and quality checks
```
Comment (muhammad-ammar) [inline Makefile]: I think this is not running quality commands.

File: .coveragerc (hunk 1 above)
Comment (muhammad-ammar) [inline .coveragerc]: Can we rename this directory(`VEDA_OS01 `) with a better name?

File: VEDA_OS01/models.py [hunk 4]
```diff
@@ -298,7 +290,7 @@ class Meta:
         get_latest_by = 'video_trans_start'
 
     def __unicode__(self):
-        return '%s' % (self.edx_id) or u''
+        return '%s'.format(self.edx_id) or u''
```
Comment (muhammad-ammar) [inline VEDA_OS01/models.py]: should we use `u'%s'.format(self.edx_id)` to return unicode object?

File: VEDA_OS01/models.py [hunk 5]
```diff
@@ -307,7 +299,7 @@ class Destination (models.Model):
     destination_nick = models.CharField('Nickname (3 Char.)', max_length=3, null=True, blank=True)
 
     def __unicode__(self):
-        return '%s' % (self.destination_name) or u''
+        return '%s'.format(self.destination_name) or u''
```
Comment (muhammad-ammar) [inline VEDA_OS01/models.py]: same comment as above.

File: reports/test [hunk 6]
```diff
@@ -0,0 +1 @@
+Ignore everything in this directory.
```
Comment (muhammad-ammar) [inline reports/test]: missing new line.

File: control/tests/test_hotstore.py [hunk 7]
```diff
@@ -29,7 +28,8 @@ def setUp(self):
 
         self.H1 = Hotstore(
             video_object=VP,
-            upload_filepath=self.upload_filepath
+            upload_filepath=self.upload_filepath,
+            video_proto=VP
```
Comment (muhammad-ammar) [inline control/tests/test_hotstore.py]: last kwarg is unnecessary. [Hotstore](https://github.com/edx/edx-video-pipeline/blob/master/control/veda_hotstore.py#L36)  does not expect this.

File: control/celeryapp.py [hunk 8]
```diff
@@ -29,10 +29,17 @@
...
+    rabbitmq_user=auth_dict['rabbitmq_user'],
+    rabbitmq_pass=auth_dict['rabbitmq_pass'],
+    rabbitmq_broker=auth_dict['rabbitmq_broker']
+)
+
+CEL_BACKEND = 'amqp://{rabbitmq_user}:{rabbitmq_pass}@{rabbitmq_broker}:5672//'.format(
+    rabbitmq_user=auth_dict['rabbitmq_user'],
+    rabbitmq_pass=auth_dict['rabbitmq_pass'],
+    rabbitmq_broker=auth_dict['rabbitmq_broker']
+)
```
Comment (muhammad-ammar) [inline control/celeryapp.py]: Both the `CEL_BROKER` and `CEL_BACKEND` are exactly same. Can we use one of them?

File: requirements.txt [hunk 9]
```diff
@@ -11,3 +11,5 @@ boto
 pyyaml
 requests==2.18.1
 celery==3.1.18
+coverage==3.7.1
+pytest==3.1.3
```
Comment (muhammad-ammar) [inline requirements.txt]: I think we are not `pytest` anywhere?

File: .coveragerc (hunk 1 above)
Comment (muzaffaryousaf) [inline .coveragerc]: VEDA directory is more like django default app with settings and urls.

Comment (muzaffaryousaf) [inline .coveragerc]: Yes but not now, maybe in future.

File: .travis.yml (hunk 2 above)
Comment (muzaffaryousaf) [inline .travis.yml]: sure.

File: Makefile (hunk 3 above)
Comment (muzaffaryousaf) [inline Makefile]: Yes, that's by purpose since there are hundreds or maybe thousand failure.

File: VEDA_OS01/models.py (hunk 4 above)
Comment (muzaffaryousaf) [inline VEDA_OS01/models.py]: sure.

File: VEDA_OS01/models.py (hunk 5 above)
Comment (muzaffaryousaf) [inline VEDA_OS01/models.py]: copy that.

File: control/celeryapp.py (hunk 8 above)
Comment (muzaffaryousaf) [inline control/celeryapp.py]: Well, i'll defer that to Greg but let's do it in next round.

File: control/tests/test_hotstore.py (hunk 7 above)
Comment (muzaffaryousaf) [inline control/tests/test_hotstore.py]: It gives warning without this.

Comment (muzaffaryousaf) [inline control/tests/test_hotstore.py]: it's expecting this as arg, if you can see the first parameter that's the video_proto.

File: requirements.txt (hunk 9 above)
Comment (muzaffaryousaf) [inline requirements.txt]: nope, that was for testing purpose.

File: control/tests/test_hotstore.py (hunk 7 above)
Comment (muhammad-ammar) [inline control/tests/test_hotstore.py]: i think we can remove `video_object=VP,` from here?

**Comment by muhammad-ammar** [comment]
lgtm 👍

**Pull request merged**